### ISSUE TITLE
Non-english dataset. Tacotron2 generated mels from "infrence" method completely wrong, but validation spectrograms while training are good.

### ISSUE THREAD

**Issue by nurjan84** [TensorSpeech/TensorFlowTTS#182]
I have a non english dataset (cyrilic). Preprocessed with "transliteration_cleaners".
So, decode_tacotron2.py generates mess, but validation spectrograms (generated mels while traning) are good. 
Tryed to call  inference method  with  the same input text from validation folder. It generates something random. Each time inference generates different random mel.

What I did wrong?

**Comment by tekinek** [comment]
@nurjan84  Sounds like cleaner mismatch on prepossessing and inference. If you use non-default or custom cleaner, make sure that this place is changed accordingly:
https://github.com/TensorSpeech/TensorFlowTTS/blob/9804aefab9dd600b0e9b70c9fc83590338988b8e/tensorflow_tts/bin/preprocess.py#L261

When you create an instance of LJSpeechProcessor for inference, make sure to pass correct cleaner name:
```
...
processor = LJSpeechProcessor(None, ["transliteration_cleaners"])
input_text = "xxxx xxx xxxx."
input_ids = processor.text_to_sequence(input_text)
input_ids = np.concatenate([input_ids, [len(symbols) - 1]], -1)
...
decoder_output, mel_outputs, stop_token_prediction, alignment_history = tacotron2.inference(
        tf.expand_dims(tf.convert_to_tensor(input_ids, dtype=tf.int32), 0),
        tf.convert_to_tensor([len(input_ids)], tf.int32),
        tf.convert_to_tensor([0], dtype=tf.int32),
)
...
```

**Comment by nurjan84** [comment]
**bin/preprocess.py**

I did preprocessing with "transliteration_cleaners"

`processor = dataset_processor[config["dataset"]](
        config["rootdir"], cleaner_names="transliteration_cleaners"
    )`

**examples/decode_tacotron2.py**
Has no any settings for cleaners. It just takes ids from validation folder and then generates mels (tacotron2.inference()). 

Evaluation sprectrograms generated during traning are good. I'm also saving mel data (not only pictures) while traning.
 
So, mels from **examples/decode_tacotron2.py** are random, but evaluation mels are good.

I tried to generate mels like in colab notebook (E2E-TensorflowTTS Demo) with _transliteration_cleaners_  and my checkpoints. It also gives me random mels.

**Comment by dathudeptrai** [comment]
@nurjan84 can you share your tensorboard and ur mel-predicted in valid ?

**Comment by nurjan84** [comment]
I'm trainig in Google Colab. Here is predicted spectrograms for same input ids.  Archive contains mel files and generated sound.
[predicted mels_and_sounds.zip](https://github.com/TensorSpeech/TensorFlowTTS/files/5028881/predicted.mels_and_sounds.zip)


Evaluation spectrograms and aligment while training at 3600 steps: It sounds good.

![1](https://user-images.githubusercontent.com/6967566/89419334-130b8180-d753-11ea-88a7-09cf1171aa67.png)

![1_alignment](https://user-images.githubusercontent.com/6967566/89419478-37675e00-d753-11ea-9317-aa9a1a919f06.png)


Predicted spectrogram from examples/decode_tacotron2.py: It looks and sounds completely different from evaluated one.
It sounds not like a speech, just "mmmmmm aaaaaaaa mmmmmm" something like this. And every prediction just new random mel with variable shape (1, xxx, 80).

![image](https://user-images.githubusercontent.com/6967566/89422371-0f79f980-d757-11ea-93b3-834e2794caf4.png)


![image](https://user-images.githubusercontent.com/6967566/89422284-f113fe00-d756-11ea-9632-57e2038c8395.png)

**Comment by tekinek** [comment]
@nurjan84 can you use following notebook to test your model?
https://github.com/TensorSpeech/TensorFlowTTS/blob/master/notebooks/tacotron2_inference.ipynb

It is good way to check your trained model step by step. You can do inference on custom text or just load the example ids from validation set. 

What is that loop in your tensorboard loss figures? (resuming?)  by the way, "mmmmmm aaaaaaaa mmmmmm" can be a normal behavior of a model at the 3600 steps.

**Comment by dathudeptrai** [comment]
@nurjan84 3600 steps is just too small. You need to train around 50k-80k steps.

**Comment by nurjan84** [comment]
The loops in loss graphs are from previous training. I was experimenting.
Previously, tacotron2 was trained 21k steps, and I got same results : validations was good, inferece was bad.
Then, I started to experiment with preprocessing code.  Deleted some punctuations symbols , that not present in my dataset.
Now its like this:
`_punctuation = ",.? "`

Ok, I will train 50k steps and then we will see whats going on )))
Thank You!

**Comment by nurjan84** [comment]
Hello I think I have found the problem)))

**tacotron2.v1.yaml** example config has no **dataset** parameter , it was setted in 
**Tacotron2Config** class as default value (dataset="ljspeech")  and vocab_size was computed according to default value.
So, all my training experimets was wrong. I was using that example tacotron2.v1.yaml file

**Comment by nurjan84** [comment]
Oh, I see that tacotron2.v1.yaml was updated in new release of tensorflowTTS.

**Comment by dathudeptrai** [comment]
the problem is solved, i will close.
